crates/matrix-sdk/src/live_location_share.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Types for live location sharing.
+
+use ruma::{
+    events::{beacon_info::BeaconInfoEventContent, location::LocationContent},
+    MilliSecondsSinceUnixEpoch, OwnedUserId,
+};
+use tokio::sync::broadcast;
+
+use crate::event_handler::EventHandlerHandle;
+
+/// Details of the last known location beacon.
+#[derive(Clone, Debug)]
+pub struct LastLocation {
+    /// The most recent location content of the user.
+    pub location: LocationContent,
+    /// The timestamp of when the location was updated.
+    pub ts: MilliSecondsSinceUnixEpoch,
+}
+
+/// Details of a users live location share.
+#[derive(Clone, Debug)]
+pub struct LiveLocationShare {
+    /// The user's last known location.
+    pub last_location: LastLocation,
+    /// Information about the associated beacon event.
+    pub beacon_info: BeaconInfoEventContent,
+    /// The user ID of the person sharing their live location.
+    pub user_id: OwnedUserId,
+}
+
+/// A subscription to live location sharing events.
+///
+/// This struct holds the `EventHandlerHandle` and the
+/// `Receiver<LiveLocationShare>` for live location shares.
+#[derive(Debug)]
+pub struct LiveLocationSubscription {
+    /// Manages the event handler lifecycle.
+    pub event_handler_handle: EventHandlerHandle,
+    /// Receives live location shares.
+    pub receiver: broadcast::Receiver<LiveLocationShare>,
+}
+
+impl Drop for LiveLocationSubscription {
+    fn drop(&mut self) {}
```

Comment:
That's where you would have to call `Client::remove_event_handler`. You can use a `EventHandlerDropGuard` though, that would be this thing automatic too. If you use https://github.com/matrix-org/matrix-rust-sdk/pull/4253, this is no longer required as you don't need to store a `receiver` and the `event_handler_handle`.